In [2]:
import pandas as pd
import numpy as np
import statsmodels.formula.api as smf

In [13]:
df = pd.read_csv("/Users/eric/ecommerce_sql_project/data/regression_data.csv")

df.head()

CustomerID  order_value  product_count  frequency  recency
0       17850       139.12              7         34      372
1       17850        22.20              2         34      372
2       13047       278.73             12         10       31
3       13047        70.05              4         10       31
4       13047        17.85              1         10       31

In [14]:
#特征工程
df['log_order_value'] = np.log1p(df['order_value'])

df.describe()

CustomerID    order_value  product_count     frequency       recency  \
count  18532.000000   18532.000000   18532.000000  18532.000000  18532.000000   
mean   15266.378211     480.865956      20.928016     18.140298     45.072469   
std     1734.116296    1678.195579      23.816071     33.979948     70.912600   
min    12346.000000       0.380000       1.000000      1.000000      0.000000   
25%    13776.250000     158.037500       6.000000      3.000000      4.000000   
50%    15178.500000     303.040000      15.000000      7.000000     17.000000   
75%    16779.000000     471.840000      27.000000     17.000000     50.000000   
max    18287.000000  168469.600000     541.000000    209.000000    373.000000   

       log_order_value  
count     18532.000000  
mean          5.592254  
std           1.054514  
min           0.322083  
25%           5.069140  
50%           5.717159  
75%           6.158757  
max          12.034517

In [15]:
#建立回归模型
model = smf.ols(
    'log_order_value ~ frequency + product_count + recency',
    data=df
).fit(cov_type='HC3')   # 使用稳健标准误

print(model.summary())

                            OLS Regression Results                            
Dep. Variable:        log_order_value   R-squared:                       0.173
Model:                            OLS   Adj. R-squared:                  0.173
Method:                 Least Squares   F-statistic:                     260.1
Date:                Thu, 26 Feb 2026   Prob (F-statistic):          2.31e-165
Time:                        20:13:18   Log-Likelihood:                -25518.
No. Observations:               18532   AIC:                         5.104e+04
Df Residuals:                   18528   BIC:                         5.108e+04
Df Model:                           3                                         
Covariance Type:                  HC3                                         
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         5.2864      0.018    298.362

回归归因结论:

1.商品种类（product_count）是订单金额的最强驱动因素（β = 0.018） 

2.最近活跃客户订单金额更高（β = -0.0011）

3.在控制商品种类和活跃度后，频次对单笔金额影响较小且略为负向